### PR TITLE
Many things

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -956,12 +956,16 @@ const ContainerService = {
 
     const userContextId = ContainerService._getUserContextIdFromTab(tab);
     return ContainerService.getIdentity({userContextId}).then(identity => {
+      const hbox = viewFor(tab.window).document.getElementById("userContext-icons");
+
       if (!identity) {
+        hbox.setAttribute("data-identity-color", "");
+        hbox.hidden = true;
```
Comment (jonathanKingston) [inline index.js]: This shouldn't be needed as the CSS takes care of this.

File: index.js [hunk 2]
```diff
@@ -956,12 +956,16 @@ const ContainerService = {
...
         return;
       }
 
-      const hbox = viewFor(tab.window).document.getElementById("userContext-icons");
       hbox.setAttribute("data-identity-color", identity.color);
+      hbox.hidden = false;
```
Comment (jonathanKingston) [inline index.js]: Same here

File: index.js (hunk 1 above)
Comment (jonathanKingston) [inline index.js]: Happy for you to remove the CSS that does this instead though :)

File: index.js [hunk 3]
```diff
@@ -784,7 +784,7 @@ const ContainerService = {
       } else {
         this._recentBrowserWindow().then(browserWin => {
           resolve(browserWin);
-        });
+        }).catch(() => {});
```
Comment (jonathanKingston) [inline index.js]: Probably should throw something like:
```
new Error("This shouldn't happen when checking for a window");
```

Or similar.... and any other catch's we have without errors

File: index.js [hunk 4]
```diff
@@ -1253,9 +1263,14 @@ ContainerWindow.prototype = {
   _configureAllTabsMenu() {
     return this._configureMenu("alltabs_containersTab", null, e => {
       const userContextId = parseInt(e.target.getAttribute("data-usercontextid"), 10);
-      ContainerService.openTab({
-        userContextId: userContextId,
-        source: "alltabs-menu"
+      ContainerService.showTabs({
```
Comment (jonathanKingston) [inline index.js]: Would it be easier to show all tabs in the openTab function so this would never happen again?

File: index.js [hunk 5]
```diff
@@ -228,15 +228,16 @@ const ContainerService = {
...
     });
 
     // Modify CSS and other stuff for each window.
 
-    this.configureWindows().catch(() => {});
+    this._configureWindows().catch(() => {});
 
     windows.browserWindows.on("open", window => {
-      this.configureWindow(viewFor(window)).catch(() => {});
+      this._configureWindow(viewFor(window)).catch(() => {});
```
Comment (jonathanKingston) [inline index.js]: Please add Errors here too.

Comment (bakulf) [inline index.js]: Yeah, we should do it for each catch(). Separate commit for it, separate PR.

**Pull request merged**